src/main/java/org/kizombadev/markdownparser/HtmlGenerator.java

```diff
@@ -0,0 +1,51 @@
+package org.kizombadev.markdownparser;
+
+import org.kizombadev.markdownparser.entities.*;
+import org.kizombadev.markdownparser.entities.interfaces.ImmutableSyntax;
+
+public class HtmlGenerator {
+
+    private ImmutableSyntax root;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "root" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the "root" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1450)